main.go

```diff
@@ -162,12 +170,31 @@ func main() {
 
 	// Detect Project ID
 	if *gcpProjectID == "" {
-		project_id, err := GetProjectIdFromMetadata()
-		if err != nil {
-			log.Fatal(err)
-		}
+		credentialsFile := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+
+		if credentialsFile != "" {
+			var svc ServiceAccount
+
+			c, err := ioutil.ReadFile(credentialsFile)
+			if err != nil {
+				log.Fatalf("Unable to read %s: %v", credentialsFile, err)
+			}
 
-		*gcpProjectID = project_id
+			json.Unmarshal(c, &svc)
+
+			if svc.ProjectId == "" {
+				log.Fatalf("Could retrieve project ID from %s", credentialsFile)
+			}
+
+			*gcpProjectID = svc.ProjectId
+		} else {
+			project_id, err := GetProjectIdFromMetadata()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			*gcpProjectID = project_id
+		}
 	}
 
 	exporter, err := NewExporter(*gcpProjectID)
```

Comment:
I realized we don't deal with the situation where , at the end of the Project ID Detection, the `*gcpProjectID` is still empty.

Do you mind adding it here as part of this PR and log.fatal if it is still empty ?

Reply:
Good catch